packages/Discretization/src/DTK_PointInCell_def.hpp

```diff
@@ -31,140 +56,76 @@ void PointInCell<DeviceType>::search(
     DTK_REQUIRE( reference_points.extent( 1 ) == physical_points.extent( 1 ) );
     DTK_REQUIRE( reference_points.extent( 1 ) == cells.extent( 2 ) );
 
-    using ExecutionSpace = typename DeviceType::execution_space;
-    int const n_ref_pts = reference_points.extent( 0 );
-
     // Perform the point in cell search. We hide the template parameters used by
     // Intrepid2, using the CellType template.
     // Note that if the Newton solver does not converge, Intrepid2 will just
     // return the last results and there is no way to know that the coordinates
     // in the reference frames where not found.
-    unsigned int const cell_topo_key = cell_topo.getKey();
-    if ( cell_topo_key ==
-         shards::getCellTopologyData<shards::Hexahedron<8>>()->key )
+    if ( cell_topo == DTK_HEX_8 )
```

Comment:
With `cell_topo` having an enumeration type, this actually becomes a great place for a `switch` statement.
You don't have to change though.